kornia/augmentation/base.py

```diff
@@ -219,8 +237,25 @@ class AugmentationBase2D(_AugmentationBase):
                                       input tensor. If ``False`` and the input is a tuple the applied transformation
                                       wont be concatenated.
         same_on_batch (bool): apply the same transformation across the batch. Default: False.
+        keepdim (bool): whether to keep the output shape the same as input (True) or broadcast it
+                        to the batch form (False). Default: False.
     """
 
+    def __check_batching__(self, input: Union[torch.Tensor,
```

Comment:
I guess this could be a class method in `_AugmentationBase`

Reply:
2D and 3D have different shapes for batching. So it would be different implementations.